lapack/gonum/dgesc2.go

```diff
@@ -0,0 +1,57 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+// Dgesc2 solves a system of linear equations
+//   A * X = scale* RHS
+// with a general N-by-N matrix A using the LU factorization with
+// complete pivoting computed by Dgetc2.
+func (impl Implementation) Dgesc2(n int, a []float64, lda int, rhs []float64, ipiv, jpiv []int) (scale float64) {
```

Comment:
Please add that the result is placed in `rhs` on exit.

Also, since this is an auxiliary routine, it's worth adding
```
//
// Dgesc2 is an internal routine. It is exported for testing purposes.
```